### ISSUE TITLE
Training, cost never decrease from 20.

### ISSUE THREAD

**Issue by juanluisrosaramos** [MaybeShewill-CV/CRNN_Tensorflow#187]
My intention is to train a model with full Synth 90k dataset and give it back to you. I also want to use your code with the latest  TF version or at least the provided in your master branch (as they corrected a bug related to log_probabilities).And I also need a more robust model that works with different text orientations so I wat to be able to train your architecture.

I try it hardly but never been able to go beyond a 20 in the cost function, it oscillates between 21 and 24. At epoch 2k it already attain the minimum cost (and a ridiculous accuracy). I try very different configurations:
- played with different LR values and decreasing policies
- different TF versions and gpus
- from 300k training images to 3 millions training tfRecords
- I went to 40k epochs but it never decrease less than the attained at 2k epoch
- retrain from an already existing snapshot

I already played with the hyperparameters without any result. Please, can you give a clue how I can surpass and minimise the 20 cost barrier?

Thankyou

**Comment by david-morris** [comment]
I'm trying to do the exact same thing.  I noticed when training a single-character dataset, the loss never goes below 5.  Did you find any other Tensorflow implementations of sequence transduction text recognition?

**Comment by liangyanfeng** [comment]
Same situation. how to resolve these problems?

**Comment by soldierofhell** [comment]
On my dataset I observe similar effect but only on "master" branch (Tensorflow 1.13). When trained on "revert-138-patch-2" (Tensorflow 1.7) the loss (at least on training dataset) gradually decreases.
I wonder if there were some important changes in model from "revert.." to current "master"? Or maybe it's something with recent versions of Tensorflow?

**Comment by soldierofhell** [comment]
I think I found a bug in batch_normalization. Current flag is set according to `is_training=self.phase == 'train'`, but ShadowNet.init still uses private attribute `self.__phase = phase` witout @phase.setter's lower()

**Comment by soldierofhell** [comment]
Empirically this fix definitely helps on convergence

**Comment by soldierofhell** [comment]
Unfortunatelly I still observe that loss gets stuck after some iterations.
It would be a good practice to compare training results for each change in repo/dependencies, because now it's hard to debug it

**Comment by MaybeShewill-CV** [comment]
@soldierofhell You may update the repo and test:)

**Comment by soldierofhell** [comment]
Thanks for the updates, I will

**Comment by MaybeShewill-CV** [comment]
@soldierofhell You're welcome

**Comment by soldierofhell** [comment]
@MaybeShewill-CV, there's a lot of changes in new version, for example in data preparation. Is this your final concept for some time or we should treat it as "experimental" and expect further changes in near future? I'm asking this kind of question because I don't see any update in README which drives me to conclusion that it's not your final statement :)

**Comment by MaybeShewill-CV** [comment]
@soldierofhell If you read README carefully you will find the difference:)

**Comment by fate233** [comment]
thankyou
